From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 1 Jun 2019 23:16:24 +0200
Subject: [PATCH] {MAYBE UPS} Use unsigned int to fix
 -Waggressive-loop-optimizations

gcc 9 warns about undefined behavior with negative indexes:

    sound/pci/echoaudio/echoaudio.c: In function 'snd_echo_vumeters_get':
    sound/pci/echoaudio/echoaudio_dsp.c:647:9: error: iteration
    1073741824 invokes undefined behavior
    [-Werror=aggressive-loop-optimizations]
      647 |   meters[n] = 0;
          |         ^
    sound/pci/echoaudio/echoaudio_dsp.c:646:2: note: within this loop
      646 |  for (; n < 32; n++)
          |  ^~~
    cc1: all warnings being treated as errors
---
 sound/pci/echoaudio/echoaudio_dsp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/pci/echoaudio/echoaudio_dsp.c b/sound/pci/echoaudio/echoaudio_dsp.c
index b181752b8481..f071ae725e15 100644
--- a/sound/pci/echoaudio/echoaudio_dsp.c
+++ b/sound/pci/echoaudio/echoaudio_dsp.c
@@ -635,7 +635,7 @@ This function assumes there are no more than 16 in/out busses or pipes
 Meters is an array [3][16][2] of long. */
 static void get_audio_meters(struct echoaudio *chip, long *meters)
 {
-	int i, m, n;
+	unsigned int i, m, n;
 
 	m = 0;
 	n = 0;
-- 
